tests/unit/broccoli/ember-app/app-and-dependencies-test.js

```diff
@@ -264,4 +272,79 @@ describe('EmberApp#appAndDependencies', function() {
       'my-addon/index.js',
     ]);
   }));
+
+  it('uses preprocessor that is marked by default', co.wrap(function *() {
+    let app = createApp();
+
+    app.registry.add('js', {
+      ext: 'js',
+      name: 'addon1',
+      isDefaultForType: true,
+      toTree(tree) {
+        return tree;
+      },
+    });
+
+    app.registry.add('js', {
+      ext: 'js',
+      name: 'addon2',
+      toTree(tree) {
+        expect(true).to.equal(false);
+        return tree;
+      },
+    });
+
+    yield buildOutput(app.addonTree());
+  }));
+
+  it('uses all registered preprocessors if none is marked by default', co.wrap(function *() {
+    let count = 0;
+    let app = createApp();
+
+    app.registry.add('js', {
+      ext: 'js',
+      name: 'addon1',
+      toTree(tree) {
+        count++;
+        return tree;
+      },
+    });
+
+    app.registry.add('js', {
+      ext: 'js',
+      name: 'addon2',
+      toTree(tree) {
+        count++;
+        return tree;
+      },
+    });
+
+    yield buildOutput(app.addonTree());
+
+    expect(count).to.equal(2);
+  }));
+
+  it('throws an exception if more than one preprocessor is marked as default', co.wrap(function *() {
+    let exceptionMessage;
+    let app = createApp();
+
+    app.registry.add('template', {
+      ext: 'hbs',
+      name: 'faulty-addon',
+      isDefaultForType: true,
+      toTree(tree) {
+        return tree;
+      },
+    });
+
+    yield co(function *() {
```

Comment:
since we can't just use `expect().to.throw()`, resorted to combo of `yield` and `co`